src/torchcodec/_core/SingleStreamDecoder.cpp

```diff
@@ -1092,13 +1092,6 @@ bool SingleStreamDecoder::canWeAvoidSeeking() const {
   // Returns true if we can avoid seeking in the AVFormatContext based on
   // heuristics that rely on the target cursor_ and the last decoded frame.
   // Seeking is expensive, so we try to avoid it when possible.
-  // Note that this function itself isn't always that cheap to call: in
-  // particular the calls to getKeyFrameIndexForPts below in approximate mode
-  // are sometimes slow.
-  // TODO we should understand why (is it because it reads the file?) and
-  // potentially optimize it. E.g. we may not want to ever seek, or even *check*
-  // if we need to seek in some cases, like if we're going to decode 80% of the
-  // frames anyway.
```

Comment:
Drive-by, my previous comment above is incorrect: `getKeyFrameIndexForPts()` in approximate mode is what is **causing** the slowness, but it's not *what* is slow. It's unfortunately more complex than that. I'll follow up with more details.